### PR TITLE
translate content/zh-cn/blog/_posts/2022-04-28-Increasing-the-security-bar-in-Ingress-NGINX/ into Chinese

### PR DESCRIPTION
**Pull request by sarazqy**
translate content/zh-cn/blog/_posts/2022-04-28-Increasing-the-security-bar-in-Ingress-NGINX/
into Chinese

Fixes  https://github.com/kubernetes/website/issues/35158

### CODE REVIEW

File: content/zh-cn/blog/_posts/2022-04-28-Increasing-the-security-bar-in-Ingress-NGINX/index.md [hunk 1]
```diff
@@ -0,0 +1,282 @@
...
+<!--
+## Meet Ingress NGINX v1.2.0 and the chrooted NGINX process
+-->
+## 了解 Ingress NGINX v1.2.0 和 chrooted NGINX 进程
+
+<!--
+One of the main challenges is that Ingress-NGINX runs the web proxy server (NGINX) alongside the Ingress
+controller (the component that has access to Kubernetes API that and that creates the `nginx.conf` file).
+-->
+主要挑战之一是 Ingress-NGINX 与 Ingress 控制器（可以访问 Kubernetes API 并创建 `nginx.conf`
```
Comment (tengqm) [inline content/zh-cn/blog/_posts/2022-04-28-Increasing-the-security-bar-in-Ingress-NGINX/index.md:58]:
```suggestion
主要挑战之一是 Ingress-NGINX 运行着 Web 代理服务器（NGINX），并与 Ingress 控制器一起运行
（后者是一个可以访问 Kubernetes API 并创建 `nginx.conf` 的组件）。
```

File: content/zh-cn/blog/_posts/2022-04-28-Increasing-the-security-bar-in-Ingress-NGINX/index.md [hunk 2]
```diff
@@ -0,0 +1,282 @@
...
+文件的组件）一起运行 Web 代理服务器（NGINX）。
+
+<!--
+So, NGINX does have the same access to the filesystem of the controller (and Kubernetes service account token, and other configurations from the container). While splitting those components is our end goal, the project needed a fast response; that lead us to the idea of using `chroot()`.
+-->
+因此，NGINX 对控制器的文件系统（和 Kubernetes 服务帐户 token，以及来自容器的其他配置）具有相同的访问权限。
```
Comment (tengqm) [inline content/zh-cn/blog/_posts/2022-04-28-Increasing-the-security-bar-in-Ingress-NGINX/index.md:64]:
```suggestion
因此，NGINX 对控制器的文件系统（和 Kubernetes 服务帐户令牌，以及容器中的其他配置）具有相同的访问权限。 
```

File: content/zh-cn/blog/_posts/2022-04-28-Increasing-the-security-bar-in-Ingress-NGINX/index.md [hunk 3]
```diff
@@ -0,0 +1,282 @@
...
+Let's take a look into what an Ingress-NGINX container looked like before this change:
+-->
+让我们看一下 Ingress-NGINX 容器在此更改之前的样子：
+
+![Ingress NGINX pre chroot](ingress-pre-chroot.png)
+
+<!--
+As we can see, the same container (not the Pod, the container!) that provides HTTP Proxy is the one that watches Ingress objects and writes the Container Volume
+-->
+正如我们所见，提供 HTTP Proxy 的同一个容器（不是 Pod，是容器！）是监视 Ingress 对象并写入 Container Volume 的容器
```
Comment (tengqm) [inline content/zh-cn/blog/_posts/2022-04-28-Increasing-the-security-bar-in-Ingress-NGINX/index.md:77]:
```suggestion
正如我们所见，用来提供 HTTP Proxy 的容器（不是 Pod，是容器！）也是是监视 Ingress
对象并将数据写入容器卷的容器。
```

File: content/zh-cn/blog/_posts/2022-04-28-Increasing-the-security-bar-in-Ingress-NGINX/index.md [hunk 4]
```diff
@@ -0,0 +1,282 @@
...
+<!--
+While this is not strictly true, to understand what was done here, it's good to understand how
+Linux containers (and underlying mechanisms such as kernel namespaces) work.
+You can read about cgroups in the Kubernetes glossary: [`cgroup`](https://kubernetes.io/docs/reference/glossary/?fundamental=true#term-cgroup) and learn more about cgroups interact with namespaces in the NGINX project article
+[What Are Namespaces and cgroups, and How Do They Work?](https://www.nginx.com/blog/what-are-namespaces-cgroups-how-do-they-work/).
+(As you read that, bear in mind that Linux kernel namespaces are a different thing from
+[Kubernetes namespaces](/docs/concepts/overview/working-with-objects/namespaces/)).
+-->
+虽然这并不完全正确，但要了解这里所做的事情，最好了解 Linux 容器（以及内核命名空间等底层机制）是如何工作的。
+你可以在 Kubernetes 词汇表中阅读有关 cgroup 的信息：[`cgroup`](https://kubernetes.io/docs/reference/glossary/?fundamental=true#term-cgroup)，
```
Comment (tengqm) [inline content/zh-cn/blog/_posts/2022-04-28-Increasing-the-security-bar-in-Ingress-NGINX/index.md:101]:
```suggestion
你可以在 Kubernetes 词汇表中阅读有关 cgroup 的信息：[`cgroup`](/zh-cn/docs/reference/glossary/?fundamental=true#term-cgroup)，
```

File: content/zh-cn/blog/_posts/2022-04-28-Increasing-the-security-bar-in-Ingress-NGINX/index.md [hunk 5]
```diff
@@ -0,0 +1,282 @@
...
+此新功能仅在 Ingress-NGINX 控制器的 v1.2.0 版本中可用。
+
+<!--
+There are two required changes in your deployments to use this feature:
+* Append the suffix "-chroot" to the container image name. For example: `gcr.io/k8s-staging-ingress-nginx/controller-chroot:v1.2.0`
+* In your Pod template for the Ingress controller, find where you add the capability `NET_BIND_SERVICE` and add the capability `SYS_CHROOT`. After you edit the manifest, you'll see a snippet like:
+-->
+要使用这个功能，在你的部署中有两个必要的改变：
+* 将后缀 "-chroot" 添加到容器镜像名称中。例如：`gcr.io/k8s-staging-ingress-nginx/controller-chroot:v1.2.0`
+* 在你的 Ingress 控制器的 Pod 模板中，找到添加功能 `NET_BIND_SERVICE` 的位置并添加功能 `SYS_CHROOT`。
```
Comment (tengqm) [inline content/zh-cn/blog/_posts/2022-04-28-Increasing-the-security-bar-in-Ingress-NGINX/index.md:126]:
```suggestion
* 在你的 Ingress 控制器的 Pod 模板中，找到添加 `NET_BIND_SERVICE` 权能的位置并添加 `SYS_CHROOT` 权能。
```

File: content/zh-cn/blog/_posts/2022-04-28-Increasing-the-security-bar-in-Ingress-NGINX/index.md [hunk 6]
```diff
@@ -0,0 +1,282 @@
...
+  编辑清单后，您将看到如下代码段：
```
Comment (tengqm) [inline content/zh-cn/blog/_posts/2022-04-28-Increasing-the-security-bar-in-Ingress-NGINX/index.md:127]:
```suggestion
  编辑清单后，你将看到如下代码段：
```

File: content/zh-cn/blog/_posts/2022-04-28-Increasing-the-security-bar-in-Ingress-NGINX/index.md [hunk 7]
```diff
@@ -0,0 +1,282 @@
...
+  - ALL
+  add:
+  - NET_BIND_SERVICE
+  - SYS_CHROOT
+```
+<!--
+If you deploy the controller using the official Helm chart then change the following setting in
+`values.yaml`:
+-->
+如果你使用官方 Helm 图表部署控制器，则在 `values.yaml` 中更改以下设置：
```
Comment (tengqm) [inline content/zh-cn/blog/_posts/2022-04-28-Increasing-the-security-bar-in-Ingress-NGINX/index.md:141]:
```suggestion
如果你使用官方 Helm Chart 部署控制器，则在 `values.yaml` 中更改以下设置：
```

File: content/zh-cn/blog/_posts/2022-04-28-Increasing-the-security-bar-in-Ingress-NGINX/index.md [hunk 8]
```diff
@@ -0,0 +1,282 @@
...
+controller:
+  image:
+    chroot: true
+```
+<!--
+Ingress controllers are normally set up cluster-wide (the IngressClass API is cluster scoped). If you manage the
+Ingress-NGINX controller but you're not the overall cluster operator, then check with your cluster admin about
+whether you can use the `SYS_CHROOT` capability, **before** you enable it in your deployment.
+-->
+Ingress 控制器通常在集群范围内设置（IngressClass API 是集群范围的）。
```
Comment (tengqm) [inline content/zh-cn/blog/_posts/2022-04-28-Increasing-the-security-bar-in-Ingress-NGINX/index.md:153]:
```suggestion
Ingress 控制器通常部署在集群作用域（IngressClass API 是集群作用域的）。
```

File: content/zh-cn/blog/_posts/2022-04-28-Increasing-the-security-bar-in-Ingress-NGINX/index.md [hunk 9]
```diff
@@ -0,0 +1,282 @@
...
+如果你管理 Ingress-NGINX 控制器但你不是整个集群的操作员，请在部署中启用它**之前**与集群管理员确认您是否可以使用 `SYS_CHROOT` 功能。
```
Comment (tengqm) [inline content/zh-cn/blog/_posts/2022-04-28-Increasing-the-security-bar-in-Ingress-NGINX/index.md:154]:
```suggestion
如果你管理 Ingress-NGINX 控制器但你不是整个集群的操作员，
请在部署中启用它**之前**与集群管理员确认你是否可以使用 `SYS_CHROOT` 权能。
```

File: content/zh-cn/blog/_posts/2022-04-28-Increasing-the-security-bar-in-Ingress-NGINX/index.md [hunk 10]
```diff
@@ -0,0 +1,282 @@
...
+}
+```
+<!--
+If you deploy this configuration, someone can call `http://website.example/randomthing` and get some listing (and access) to the whole filesystem of the Ingress controller.
+
+Now, can you spot the difference between chrooted and non chrooted Nginx on the listings below?
+-->
+如果你部署了这种配置，有人可以调用 `http://website.example/randomthing` 并获取对 Ingress 控制器的整个文件系统的一些列表（和访问权限）。
+
+现在，你能在下面的列表中发现 chrooted 和非 chrooted Nginx 之间的区别吗？
```
Comment (tengqm) [inline content/zh-cn/blog/_posts/2022-04-28-Increasing-the-security-bar-in-Ingress-NGINX/index.md:178]:
```suggestion
现在，你能在下面的列表中发现 chroot 处理过和未经 chroot 处理过的 Nginx 之间的区别吗？
```

File: content/zh-cn/blog/_posts/2022-04-28-Increasing-the-security-bar-in-Ingress-NGINX/index.md [hunk 11]
```diff
@@ -0,0 +1,282 @@
...
+
+| Without extra `chroot()`             | With extra `chroot()` |
```
Comment (tengqm) [inline content/zh-cn/blog/_posts/2022-04-28-Increasing-the-security-bar-in-Ingress-NGINX/index.md:180]:
```suggestion
| 不额外调用 `chroot()`             | 额外调用 `chroot()` |
```

File: content/zh-cn/blog/_posts/2022-04-28-Increasing-the-security-bar-in-Ingress-NGINX/index.md [hunk 12]
```diff
@@ -0,0 +1,282 @@
...
+| `usr`                      | `usr`  |
+| `var`                      | `var`  |
+| `dbg`                      |        |
+| `nginx-ingress-controller` |        |
+| `wait-shutdown`            |        |
+
+<!--
+The one in left side is not chrooted. So NGINX has full access to the filesystem. The one in right side is chrooted, so a new filesystem with only the required files to make NGINX work is created.
+-->
+左侧的那个没有 chrooted。 所以 NGINX 可以完全访问文件系统。右侧的那个是 chrooted，因此创建了一个新文件系统，其中只有使 NGINX 工作所需的文件。
```
Comment (tengqm) [inline content/zh-cn/blog/_posts/2022-04-28-Increasing-the-security-bar-in-Ingress-NGINX/index.md:206]:
```suggestion
左侧的那个没有 chroot 处理。所以 NGINX 可以完全访问文件系统。右侧的那个经过 chroot 处理，
因此创建了一个新文件系统，其中只有使 NGINX 工作所需的文件。
```

File: content/zh-cn/blog/_posts/2022-04-28-Increasing-the-security-bar-in-Ingress-NGINX/index.md [hunk 13]
```diff
@@ -0,0 +1,282 @@
...
+我们知道新的 `chroot()` 机制有助于解决部分风险，但仍然有人可以尝试注入命令来读取，例如 `nginx.conf` 文件并提取敏感信息。
+
+<!--
+So, another change in this release (this is opt-out!) is the _deep inspector_.
+We know that some directives or regular expressions may be dangerous to NGINX, so the deep inspector
+checks all fields from an Ingress object (during its reconciliation, and also with a
+[validating admission webhook](/docs/reference/access-authn-authz/admission-controllers/#validatingadmissionwebhook))
+to verify if any fields contains these dangerous directives.
+-->
+所以，这个版本的另一个变化（这是选择退出！）是_deep inspector_。
```
Comment (tengqm) [inline content/zh-cn/blog/_posts/2022-04-28-Increasing-the-security-bar-in-Ingress-NGINX/index.md:226]:
```suggestion
所以，这个版本的另一个变化（可选择取消）是 **深度探测（Deep Inspector）**。
```

File: content/zh-cn/blog/_posts/2022-04-28-Increasing-the-security-bar-in-Ingress-NGINX/index.md [hunk 14]
```diff
@@ -0,0 +1,282 @@
...
+我们知道某些指令或正则表达式可能对 NGINX 造成危险，因此深度检查器会检查 Ingress 对象中的所有字段
```
Comment (tengqm) [inline content/zh-cn/blog/_posts/2022-04-28-Increasing-the-security-bar-in-Ingress-NGINX/index.md:227]:
```suggestion
我们知道某些指令或正则表达式可能对 NGINX 造成危险，因此深度探测器会检查 Ingress 对象中的所有字段
```

File: content/zh-cn/blog/_posts/2022-04-28-Increasing-the-security-bar-in-Ingress-NGINX/index.md [hunk 15]
```diff
@@ -0,0 +1,282 @@
...
+（在其协调期间，并且还使用 [验证准入 webhook](/zh-cn/docs/reference/access-authn-authz/admission-controllers/#validatingadmissionwebhook))
```
Comment (tengqm) [inline content/zh-cn/blog/_posts/2022-04-28-Increasing-the-security-bar-in-Ingress-NGINX/index.md:228]:
```suggestion
（在其协调期间，并且还使用[验证准入 webhook](/zh-cn/docs/reference/access-authn-authz/admission-controllers/#validatingadmissionwebhook))
```

File: content/zh-cn/blog/_posts/2022-04-28-Increasing-the-security-bar-in-Ingress-NGINX/index.md [hunk 16]
```diff
@@ -0,0 +1,282 @@
...
+验证是否有任何字段包含这些危险指令。
+
+<!--
+The ingress controller already does this for annotations, and our goal is to move this existing validation to happen inside
+deep inspection as part of a future release.
+-->
+入口控制器已经为注释做了这个工作，我们的目标是把现有的验证转移到深度检查中，作为未来版本的一部分。
```
Comment (tengqm) [inline content/zh-cn/blog/_posts/2022-04-28-Increasing-the-security-bar-in-Ingress-NGINX/index.md:235]:
```suggestion
Ingress 控制器已经通过注解做了这个工作，我们的目标是把现有的验证转移到深度探测中，作为未来版本的一部分。
```

File: content/zh-cn/blog/_posts/2022-04-28-Increasing-the-security-bar-in-Ingress-NGINX/index.md [hunk 17]
```diff
@@ -0,0 +1,282 @@
...
+
+<!--
+You can take a look into the existing rules in [https://github.com/kubernetes/ingress-nginx/blob/main/internal/ingress/inspector/rules.go](https://github.com/kubernetes/ingress-nginx/blob/main/internal/ingress/inspector/rules.go).
+-->
+您可以在 [https://github.com/kubernetes/ingress-nginx/blob/main/internal/ingress/inspector/rules.go](https://github.com/kubernetes/ingress-nginx/blob/main/internal/ingress/inspector/rules.go) 中查看现有规则。
```
Comment (tengqm) [inline content/zh-cn/blog/_posts/2022-04-28-Increasing-the-security-bar-in-Ingress-NGINX/index.md:240]:
```suggestion
你可以在 [https://github.com/kubernetes/ingress-nginx/blob/main/internal/ingress/inspector/rules.go](https://github.com/kubernetes/ingress-nginx/blob/main/internal/ingress/inspector/rules.go) 中查看现有规则。
```

File: content/zh-cn/blog/_posts/2022-04-28-Increasing-the-security-bar-in-Ingress-NGINX/index.md [hunk 18]
```diff
@@ -0,0 +1,282 @@
...
+
+<!--
+Due to the nature of inspecting and matching all strings within relevant Ingress objects, this new feature may consume a bit more CPU. You can disable it by running the ingress controller with the command line argument `--deep-inspect=false`.
+-->
+由于检查和匹配相关 Ingress 对象中的所有字符串的性质，此新功能可能会消耗更多 CPU。你可以通过使用命令行参数 `--deep-inspect=false` 运行入口控制器来禁用它。
```
Comment (tengqm) [inline content/zh-cn/blog/_posts/2022-04-28-Increasing-the-security-bar-in-Ingress-NGINX/index.md:245]:
```suggestion
由于检查和匹配相关 Ingress 对象中的所有字符串的性质，此新功能可能会消耗更多 CPU。
你可以通过使用命令行参数 `--deep-inspect=false` 运行 Ingress 控制器来禁用它。
```

**Comment by sarazqy** [comment]
open in  https://github.com/kubernetes/website/pull/35526

**Pull request closed without merging**